### ISSUE TITLE
Doesn't work on Safari browser

### ISSUE THREAD

**Issue by affanshahab** [JohnMealy23/cookie-toss#23]
It seems broken on Safari browser, doesn't work as expected

**Comment by JohnMealy23** [comment]
@affanshahab - Please include the error message you're seeing, along with stack trace, browser version and any other relevant information.

**Comment by zehawki** [comment]
All cross-domain local storage access is disabled by default with Safari 7+. This is a result of the "Block cookies and other website data" privacy setting being set to "From third parties and advertisers". Any cross-storage client code will not crash, however, it will only have access to a sandboxed, isolated local storage instance. As such, none of the data previously set by other origins will be accessible.

**Comment by JohnMealy23** [comment]
Because the iframe is scoped to the domain on which it's accessing localStorage, this wouldn't be a cross-domain accessing of localStorage.  I just did a test on Safari 14, and am seeing a successful suite of tests.  

@zehawki - did you test and find issue, or are you conjecturing?  Please let me know if it looks like I'm missing something.

![image](https://user-images.githubusercontent.com/1444817/100416690-12259500-304d-11eb-9ad4-9a6b74b6faab.png)

**Comment by zehawki** [comment]
>did you test and find issue, or are you conjecturing? Please let me know if it looks like I'm missing something.

Neither actually. I dont have access to Safari so I cant test firsthand, but you can see the above in Apple literature + in various posts around the internet (so its not conjecture per se...). I've started using cross-storage and you can see they mention the above quoted text here https://github.com/zendesk/cross-storage#compatibility

**Comment by zehawki** [comment]
More resources:

1. https://webkit.org/blog/8124/introducing-storage-access-api/
2. https://stackoverflow.com/questions/38584273/local-storage-cross-domain-safari-disables-it-by-default
3.  https://stackoverflow.com/questions/63922558/safari-localstorage-not-shared-between-iframes-hosted-on-same-domain
4. https://github.com/krakenjs/post-robot/issues/37
5. https://developer.apple.com/forums/thread/109909

TL;DR - seems it will work in Safari only if the user changes the default settings and thats not a dependable solution. Perhaps you have more permissive settings in your browser?

**Comment by JohnMealy23** [comment]
This definitely looks like it's a thing.  I still have yet to reproduce this - even after ensuring my Safari privacy settings are set to "Prevent cross-site tracking".  If you are able to reproduce this, I'd love to see some error messages or steps to reproduce.  I'll keep trying, in the mean time.
